backend/src/Squidex.Domain.Apps.Entities/Contents/Queries/ContentQueryService.cs

```diff
@@ -103,6 +103,13 @@ public async Task<IResultList<IEnrichedContentEntity>> QueryAsync(Context contex
                     contents = contents.SortSet(x => x.Id, q.Ids);
                 }
 
+                if (context.Permissions.Any(p => p.Id == Permissions.AppContentReadOwn))
+                {
+                    var claim = context.User.Claims.Where(p => p.Type == "sub").First().Value;
+                    var contentsList = contents.Where(c => c.CreatedBy.Identifier == claim).ToList();
+                    contents = ResultList.Create(contentsList.Count, contentsList);
```

Comment:
This would filter the contents after they are received from MongoDB, we have to do it as part of the query...

Reply:
To get  contents we use following query method: `var contents = await contentRepository.QueryAsync(context.App, schema, q, context.Scope());`

I thought to change params Q query.  Originally I wanted to do so.  You told me not to change  Filter. 

Can you show me example how to do it in query.

Reply:
What I meant is not to add a parameter to q.ClrQuery.Filter, but you can add a property to Q.

Reply:
Check my last comment. There is one issue. MongoDB.Driver for filter `filters.Add(Filter.Eq(x => x.CreatedBy.Identifier, query.CreatedBy));` displays error:  **Unable to determine the serialization information for x =\u003E x.CreatedBy.Identifier.**

Reply:
Ref token is just serialized to string. 

https://github.com/Squidex/squidex/blob/master/backend/src/Squidex.Infrastructure.MongoDb/MongoDb/RefTokenSerializer.cs

You should add a ref token to `Q` and create this with `q.CreatedBy = new RefToken(RefTokenType.User, user.OpenIdSubject())`

And then a simple filter:

 ```
filters.Add(Filter.Eq(x => x.CreatedBy, query.CreatedBy));
```

Reply:
There is an extension method for user: https://github.com/Squidex/squidex/blob/835274396f58f9b9f8f11bab8956639337a97a5b/backend/src/Squidex.Infrastructure/Security/Extensions.cs#L16

Use this please.